test/OrchardCore.Tests/Modules/OrchardCore.Resources/SubResourceIntegrityTests.cs

```diff
@@ -0,0 +1,73 @@
+using System.Security.Cryptography;
+using OrchardCore.ResourceManagement;
+using OrchardCore.Resources;
+
+namespace OrchardCore.Tests.Modules.OrchardCore.Resources
+{
+    public class SubResourceIntegrityTests
+    {
+        [Fact]
+        public void CheckSubResourceIntegrity()
+        {
+            // Arrange
+            var resourceOptions = Options.Create(new ResourceOptions());
+            var httpContextAccessorMock = new Mock<IHttpContextAccessor>();
+            httpContextAccessorMock
+                .Setup(a => a.HttpContext)
+                .Returns(new DefaultHttpContext());
+            var configurationOptions = new ResourceManagementOptionsConfiguration(
+                resourceOptions,
+                Mock.Of<IHostEnvironment>(),
+                httpContextAccessorMock.Object
+            );
+            var resourceManagementOptions = new ResourceManagementOptions();
+
+            // Act & Assert
+            configurationOptions.Configure(resourceManagementOptions);
+
+            var resourceManifest = resourceManagementOptions.ResourceManifests.First();
+
+            ValidateSubResourceIntegrity("script");
+            ValidateSubResourceIntegrity("style");
+
+            void ValidateSubResourceIntegrity(string resourceType)
+            {
+                foreach (var resource in resourceManifest.GetResources(resourceType))
+                {
+                    foreach (var resourceDefinition in resource.Value)
+                    {
+                        if (!String.IsNullOrEmpty(resourceDefinition.CdnIntegrity) && !String.IsNullOrEmpty(resourceDefinition.UrlCdnDebug))
+                        {
+                            var resourceIntegrity = GetSubResourceIntegrityAsync(resourceDefinition.UrlCdnDebug).Result;
+
+                            Assert.True(resourceIntegrity.Equals(resourceDefinition.CdnDebugIntegrity), $"The {resourceType} {resourceDefinition.UrlCdnDebug} has invalid SRI hash, please use '{resourceIntegrity}' instead.");
+                        }
+
+                        if (!String.IsNullOrEmpty(resourceDefinition.CdnIntegrity) && !String.IsNullOrEmpty(resourceDefinition.UrlCdn))
+                        {
+                            var resourceIntegrity = GetSubResourceIntegrityAsync(resourceDefinition.UrlCdn).Result;
+
+                            Assert.True(resourceIntegrity.Equals(resourceDefinition.CdnIntegrity), $"The {resourceType} {resourceDefinition.UrlCdn} has invalid SRI hash, please use '{resourceIntegrity}' instead.");
+                        }
+                    }
+                }
+            }
+        }
+
+        private static async Task<string> GetSubResourceIntegrityAsync(string url)
+        {
+            var client = new HttpClient
```

Comment:
`HttpClient` [mustn't be instantiated frequently](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-net-http-httpclient#instancing).

Reply:
The following error shows up when we use the same instance

**System.InvalidOperationException : This instance has already started one or more requests. Properties can only be modified before sending the first**

Reply:
Now the tests passed after I revert the last changes, you could check the logs

Reply:
Please fix that first.

Reply:
I already did then revert the changes when the build failed. Let me do it again

Reply:
This is still the case.

Reply:
There's something strange when using the await in this case, I need to check again

Reply:
@Piedone the reason why the test failed after using static is

https://github.com/dotnet/corefx/blob/b8b81a66738bb10ef0790023598396861d92b2c4/src/System.Net.Http/src/System/Net/Http/HttpClient.cs#L665-L672

Reply:
Please don't request a review until this is addressed. Check out the docs I linked.

Reply:
Until this works without instantiating `HttpClient` repeatedly, your code needs to be adjusted.

Reply:
You get this because the code sets the `BaseAddress` on `HttpClient`. You can't change this on an existing instance. However, you don't need to do this anyway, because in `GetByteArrayAsync()` you set the URL too. You only need to set it there, and reuse the instance.

`HttpClient` can, of course, open multiple different requests with the same instance, but not if you set `BaseAddress`.